demos/common/sample_azure_common.h

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+/**
+ * @brief Contains helpers that are used in multiple samples
+ */
+
+#ifndef SAMPLE_AZURE_COMMON_IF_H
+#define SAMPLE_AZURE_COMMON_IF_H
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+#if NDEBUG
+    #define panic_handler()                                                                           \
+    vLoggingPrintf( "[ERROR] [%s:%s:%d]\r\nRestarting device...\r\n", __FILE__, __func__, __LINE__ ); \
+    configRestartDevice()
```

Comment:
consider using a `do {} while (0)` with multiple function calls in a macro: https://stackoverflow.com/questions/2687569/c-do-while0

```suggestion
    #define panic_handler()                                                                           \
    do { vLoggingPrintf( "[ERROR] [%s:%s:%d]\r\nRestarting device...\r\n", __FILE__, __func__, __LINE__ ); \
    configRestartDevice(); } while (0)
```